Tests/NIOCoreTests/AsyncSequences/NIOThrowingAsyncSequenceTests.swift

```diff
@@ -463,6 +463,32 @@ final class NIOThrowingAsyncSequenceProducerTests: XCTestCase {
             XCTAssertTrue(error is CancellationError)
         }
     }
+    
+    @available(*, deprecated, message: "tests the deprecated custom generic failure type")
+    func testTaskCancel_whenStreaming_andSuspended_withCustomErrorType() async throws {
+        struct CustomError: Error {}
+        // We are registering our demand and sleeping a bit to make
+        // sure our task runs when the demand is registered
+        let backPressureStrategy = MockNIOElementStreamBackPressureStrategy()
+        let delegate = MockNIOBackPressuredStreamSourceDelegate()
+        let new = NIOThrowingAsyncSequenceProducer.makeSequence(
+            elementType: Int.self,
+            failureType: CustomError.self,
+            backPressureStrategy: backPressureStrategy,
+            delegate: delegate
+        )
+        let sequence = new.sequence
```

Comment:
Can we make sure to use `withExtendedLifetime` to preserve the producer side? That'll help me be confident that this code doesn't have this effect because the producer is being dropped.

Reply:
Added in https://github.com/apple/swift-nio/pull/2401/commits/c7863f6f89974138bc44d50a6a7d5932c283ea3e
We can't wrap the whole thing in `withExtendedLifetime` because we call an async method and the closure `withExtendedLifetime` takes isn't async. (another use case for reasync)

Reply:
Thanks. I'm happy with the patch as written. More broadly, if you ever want to say "I need a value to live to this point in the code", the idiom has ended up becoming to just write `withExtendedLifetime(theValue) { }`. Your version works too, so I'm happy with this as-is, but just an FYI for the future that you could have simply written that at the bottom of the file.

Reply:
Ah, nice. Thank you!